### ISSUE TITLE
Text variable anchors are not properly updated when text-radial-offset changes

### ISSUE THREAD

**Issue by stevage** [mapbox/mapbox-gl-js#10891]
**mapbox-gl-js version**: 2.3.1

**browser**: MacOS FF

### Steps to Trigger Behavior

1. Create a symbol layer with text-variable-anchor ['left','top'] and text-radial-offset 1, such that a preferred placement (left) is generally not used and `top` is.
2. Increase text-radial-offset to 3.

Expectation: anchors recalculated so that `left` is generally used now.

Actually: `top` is still used everywhere, just further away.

You can force it to recalculate only by modifying `text-variable-anchor` so that the previous anchor (`top`) is not possible anymore.

<img width="1349" alt="Screenshot 2021-07-22 at 2 29 34 PM" src="https://user-images.githubusercontent.com/678344/126590310-1e324070-ed94-4a6c-8d74-f8363c7df905.png">
<img width="1238" alt="Screenshot 2021-07-22 at 2 30 07 PM" src="https://user-images.githubusercontent.com/678344/126590318-26a2dff9-93a3-42a6-aef4-6ae0c70d9f0c.png">
<img width="1247" alt="Screenshot 2021-07-22 at 2 30 25 PM" src="https://user-images.githubusercontent.com/678344/126590322-9c3c8869-c614-49ea-b468-c642904c8d55.png">
<img width="1236" alt="Screenshot 2021-07-22 at 2 30 37 PM" src="https://user-images.githubusercontent.com/678344/126590325-789df33b-a9cc-4a3a-85ea-2f624e1a1fad.png">

The result after the 4th image is what I expect to see in the 2nd image.

**Comment by rreusser** [comment]
Thanks for filing the issue, @stevage. I had to dig a little, but I believe it's the `placedOrientation` of field of symbols that stores the current placement state. I believe the reason for this behavior is to minimize visual changes when interacting with the map:

![text-variable-anchor](https://user-images.githubusercontent.com/572717/127194751-f8339c75-c2d9-4e1c-9fc1-43f49fac5ccf.gif)

In other words, it's maybe like a one-way ratchet? Variable-placement symbols will improve placement if a label can't be displayed, but they won't revert to a more preferred placement if both placements are once again equally viable.

This behavior seems defensible, but perhaps what you've noticed is that this is persistent even when the style properties controlling it are modified. I wonder if it could flush the persistent state when the properties is modified.

In short, I think I see and can reproduce what you're looking at, but if you'll permit me to ask, can you elaborate on your goal? I can imagine sticky state like this producing frustration while designing a map, and that's worth thinking about how to improve. Does it result in runtime issues for the end-user experience?

**Comment by stevage** [comment]
Yeah, I had sort of guessed it was a deliberate choice to avoid distracting visual changes. But yes, I think the state should be flushed when the style changes.

My goal? I don't really use Studio - I do all my style tweaking in JavaScript, often in the console. And I, hopefully not unreasonably, expect that any call to `setLayoutProperty` actually changes the style in the way it promises.

> Does it result in runtime issues for the end-user experience?

Not directly.
